smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/EventStreamGenerator.java

```diff
@@ -0,0 +1,177 @@
+            LanguageTarget target
+    ) {
+        if (!hasEventStream(model, settings.getService(model))) {
+            return;
+        }
+
+        switch (target) {
+            case NODE:
+                writer.addDependency(TypeScriptDependency.AWS_SDK_EVENTSTREAM_SERDE_NODE);
+                writer.addImport(
+                        "eventStreamSerdeProvider",
+                        "eventStreamSerdeProvider",
+                        TypeScriptDependency.AWS_SDK_EVENTSTREAM_SERDE_NODE.packageName
+                );
+                writer.write("eventStreamSerdeProvider");
+                break;
+            case BROWSER:
+                writer.addDependency(TypeScriptDependency.AWS_SDK_EVENTSTREAM_SERDE_BROWSER);
+                writer.addImport(
+                        "eventStreamSerdeProvider",
+                        "eventStreamSerdeProvider",
+                        TypeScriptDependency.AWS_SDK_EVENTSTREAM_SERDE_BROWSER.packageName
+                );
+                writer.write("eventStreamSerdeProvider");
+                break;
+            default:
+                // do nothing
+        }
+    }
+
+    public static final boolean hasEventStream(
+            Model model,
+            ServiceShape service
+    ) {
+        TopDownIndex topDownIndex = model.getKnowledge(TopDownIndex.class);
+        Set<OperationShape> operations = topDownIndex.getContainedOperations(service);
+        for (OperationShape operation : operations) {
+            if (operationHasEventStream(model, service, operation)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public static final boolean operationHasEventStream(
+            Model model,
+            ServiceShape service,
+            OperationShape operation
+    ) {
+        if (operationHasEventStreamInput(model, operation) || operationHasEventStreamOutput(model, operation)) {
+            return true;
+        }
+        return false;
+    }
+
+    public static final boolean operationHasEventStreamInput(
+            Model model,
+            OperationShape operation
+    ) {
+        EventStreamIndex eventStreamIndex = model.getKnowledge(EventStreamIndex.class);
+        if (eventStreamIndex.getInputInfo(operation).isPresent()) {
+            return true;
+        }
+        return false;
+    }
+
+    public static final boolean operationHasEventStreamOutput(
+            Model model,
+            OperationShape operation
+    ) {
+        EventStreamIndex eventStreamIndex = model.getKnowledge(EventStreamIndex.class);
+        if (eventStreamIndex.getOutputInfo(operation).isPresent()) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * The value of event header 'type' property of given shape.
+     *
+     * @param shape shape bond to event header
+     * @return string literal for the event message header type
+     */
+    public static String getEventHeaderType(Shape shape) {
```

Comment:
This looks like it's only used in one place, which isn't in this class. It can probably be moved there and be made private.